### ISSUE TITLE
Where does the code reflect the "Pruning Backdoor by Importance Driven Cloning" in the paper？

### ISSUE THREAD

**Issue by bddk520** [qiulingxu/MEDIC#1]
In the paper, the implementation of the MEDIC method requires the completion of Cloning and Pruning Backdoor by Importance Driven Cloning. But when I read the code, I only saw the part that implements Cloning. I did not see the code for Pruning Backdoor by Importance Driven Cloning. I have doubts about the code implementation in the official repository.This may be because my understanding of the code is not clear enough. Hope it can be explained.

**Comment by qiulingxu** [comment]
Hi, bddk520, Thanks for the question. TLDR: we don't have pruning phase. 

Basically, how the removal works happens in the way of assigning weight to important and impactful neurons. Neurons that are not related to the normal behaviors are assigned with small weight compared to average during cloning. It is highly likely that backdoor neurons fall into such category.  Such cloning loss is then combined with normal training loss. As long as cloning loss is properly weighted, we can remove the backdoor behavior without further pruning.
